### ISSUE TITLE
gc-merge-devnet-1 TBD-issue Jul 3

### ISSUE THREAD

**Issue by dapplion** [gnosischain/consensus-deployment-ansible#9]
------

Nethermind ([user list](https://github.com/gnosischain/consensus-deployment-ansible/blob/6d9bbf6dde9abb6501abdd2fa3c6d4eb75dc7394/gc-merge-devnet-0/inventory/group_vars/eth1client_nethermind.yml#L8)) and Lighthouse ([user list](https://github.com/gnosischain/consensus-deployment-ansible/blob/6d9bbf6dde9abb6501abdd2fa3c6d4eb75dc7394/gc-merge-devnet-0/inventory/group_vars/eth2client_lighthouse.yml#L10)) devs can ssh to hosts ([ip list here](https://github.com/gnosischain/consensus-deployment-ansible/blob/master/gc-merge-devnet-0/inventory/inventory.ini)) `ssh devops@<ip>`. Same as Pari's deployments

------

- gc-merge-devnet-1 was launched at Jun 30 2022 12:05:00 GMT+0000 https://github.com/gnosischain/consensus-deployment-ansible/pull/7. 10 validating clients, diversity on CL; EL all nethermind, each has 1000 keys, network of 10000 validators total. The first two nodes are POSDAO validators.
  - Beaconchain explorer http://143.198.151.36:3333/
  - Blockscout explorer http://143.198.151.36:3334/
- TTD was reached at Jun 30 2022 13:56:30 GMT+0000
- At Jul 2 2022 20:00:00 GMT+0000 participation dropped to 90%, as lodestar-nethermind-1 started to experience issues
- At Jul 4 2022 15:00:00 GMT+0000 participation dropped to 60%. Some nodes CL or/and EL stopped progressing and lost all peers

![Screenshot from 2022-07-05 13-21-01](https://user-images.githubusercontent.com/35266934/177316491-9fe65345-13e8-42ba-8575-93c8d6561556.png)
![Screenshot from 2022-07-05 13-21-35](https://user-images.githubusercontent.com/35266934/177316604-038564b8-29ef-478a-ab8a-ac573d763232.png)
![Screenshot from 2022-07-05 13-21-50](https://user-images.githubusercontent.com/35266934/177316609-1a93988e-fdb1-4e53-af62-44adc4451f03.png)
![Screenshot from 2022-07-05 13-22-11](https://user-images.githubusercontent.com/35266934/177316679-eebb571b-4f5c-4de4-8590-f40419881dcf.png)
![Screenshot from 2022-07-05 13-22-24](https://user-images.githubusercontent.com/35266934/177316688-8fdb1515-c0a4-4527-866f-dcb5cc576614.png)

-----

**NOTE**: Currently digging in the logs, will post updates

**Comment by dapplion** [comment]
lodestar-nethermind-1 was the first host to experience issues. The beacon node (lodestar) stalled at ~ Jul 3 2022 6:00:00 GMT+0000

![Screenshot from 2022-07-05 18-08-24](https://user-images.githubusercontent.com/35266934/177370881-747891aa-1f9e-468f-96ce-d982c6cb4da5.png)

**Comment by dapplion** [comment]
`lodestar-nethermind-1` validator performance clearly suffered due to the communication issues with EL and incapacity to follow the head. This cause the participation drop observed in the network.

- But why did the host performance started to degrade in the first place?

![Screenshot from 2022-07-06 23-51-57](https://user-images.githubusercontent.com/35266934/177649817-6d6eda80-2981-4067-ad86-b251d0799c5a.png)

**Comment by dapplion** [comment]
@michaelsproul

Recurring logs that happen extremely often after ~ Jul-03 00:00:00.

```
Jul 03 15:42:30.547 ERRO Failure verifying sync committee signature for gossip, validator_index: 2872, slot: 54449, request_index: 0, error: PastSlot { message_slot: Slot(54449), earliest_permissible_slot: Slot(
54450) }
...
Jul 03 18:19:51.079 ERRO Failure verifying signed contribution and proof, contribution_slot: 56337, subcommittee_index: 1, aggregator_index: 2222, request_index: 2, error: PastSlot { message_slot: Slot(56337), e
arliest_permissible_slot: Slot(56338) }
```

Some notable logs I found. For temporal reference lighthouse-nethermind-3 stalls at Jul-04 15:00:00.

![Screenshot from 2022-07-07 16-45-07](https://user-images.githubusercontent.com/35266934/177802422-cbb0a907-7786-4a12-8301-cb033e8eb60b.png)


```
Jul 04 14:47:25.918 DEBG Execution engine call failed            id: http://127.0.0.1:8560/, error: Reqwest(reqwest::Error { kind: Request, url: Url { scheme: "http", cannot_be_a_base: false, username: "", passw
ord: None, host: Some(Ipv4(127.0.0.1)), port: Some(8560), path: "/", query: None, fragment: None }, source: TimedOut }), service: exec
```

**Comment by michaelsproul** [comment]
> lighthouse seems to be downscoring peers as a result of the inn-ability to validate blocks with the EL?

This was an issue in v2.3.0 but has since been fixed in this commit which is part of v2.3.2-rc.0: https://github.com/sigp/lighthouse/commit/f4287197619b6fcf280ce34e9d778573f641e279. I'd suggest running the latest release (candidate) or even just `unstable` directly for these nets seeing as we're at the cutting edge.

> That means that the communication between CL-EL is okay right? There no explicit log about the EL responding ok.

Yeah that's right, although we should maybe add a debug log

> why does lighthouse go looking for a block Searching for block: 0xe28d…a9fe immediately making it head? New block received root: 0xe28d7dbcd2acf..

There's a mostly harmless race where we'll go looking for a block on RPC if we haven't seen it on gossip, at around the same time as it arrives on gossip. This is improved by a new PR we haven't merged yet: https://github.com/sigp/lighthouse/pull/3317

> Note that the snooper does not acknowledge any request at Jul 04 14:58:30.835 WARN Error whilst processing payload status error: Offline { id: "http://127.0.0.1:8560/" }, service: exec Did Lighthouse attempted to contact the EL?

Lighthouse caches the offline status, and will do an "upcheck" (hitting `eth_syncing`) before properly sending the request. If the upcheck fails then the actual request won't get sent. This is mostly a hangover from the redundant EE design, and we're working on simplifying it now that we only support a single EE. This is the tracking issue (which was relevant to the last devnet explosion too): https://github.com/sigp/lighthouse/issues/3176

**Comment by dapplion** [comment]
Closing stale issue
